src/main/java/Duke.java

```diff
@@ -1,10 +1,59 @@
+import java.util.Scanner;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * A chatbot that helps the user take note of tasks.
+ *
+ * @author Armando Jovan Kusuma
+ */
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private Scanner scanner;
+
+    /**
+     * Constructs an instance of the chat bot.
+     */
+    public Duke() {
+        this.scanner = new Scanner(System.in);
+        this.ui = new Ui();
+        this.storage = new Storage();
+    }
+    /**
+     * Runs the Duke chatbot.
+     */
+    public void run() {
```

Comment:
Perhaps you can consider adding line breaks in the method body to separate logical blocks and improve readability.